pkg/codec/property_extractor.go

```diff
@@ -0,0 +1,98 @@
+package codec
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+// NewPropertyExtractor creates a modifier that will extract a single property from a struct.
+// This modifier is lossy, as TransformToOffchain will discard unwanted struct properties and
+// return a single element. Calling TransformToOnchain will result in unset properties.
+func NewPropertyExtractor(fieldName string) Modifier {
+	m := &propertyExtractor{
+		modifierBase: modifierBase[bool]{
+			fields:           map[string]bool{fieldName: false},
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+		fieldName: fieldName,
+	}
+
+	m.modifyFieldForInput = func(_ string, field *reflect.StructField, _ string, _ bool) error {
+		return nil
+	}
+
+	return m
+}
+
+type propertyExtractor struct {
+	modifierBase[bool]
+	fieldName string
+}
+
+func (e *propertyExtractor) TransformToOnChain(offChainValue any, _ string) (any, error) {
+	parts := strings.Split(e.fieldName, ".")
+	fieldName := parts[len(parts)-1]
+
+	fields := []reflect.StructField{
+		{Name: fieldName, Type: reflect.TypeOf(offChainValue)},
+	}
```

Comment:
Consider that the way modifiers work today, they could extract 

A.B.C and expect to get C on-chain.  The off-chain would need to nest C correctly on the way back to be an A with a B with a C not just a B with a C